src/uct/tcp/tcp_cm.c

```diff
@@ -165,9 +205,10 @@ ucs_status_t uct_tcp_cm_send_event(uct_tcp_ep_t *ep,
     ucs_assertv(!(event & ~(UCT_TCP_CM_CONN_REQ |
                             UCT_TCP_CM_CONN_ACK)),
                 "ep=%p", ep);
-    ucs_assertv(!(ep->flags & UCT_TCP_EP_FLAG_CTX_TYPE_TX) ||
```

Comment:
why removed assert?

Reply:
to simplify code in `uct_tcp_cm_handle_conn_req()`, RX is set before handling `if (ep->conn_state == UCT_TCP_EP_CONN_STATE_CONNECTED)` and sending ACK.
so, sending ACK may trigger this assertions, since both RX/TX are set and EP is connected